fetchai/ledger/api/__init__.py

```diff
@@ -89,8 +98,10 @@ def sync(self, txs: Transactions, timeout=None):
                 failures = ['{}:{}'.format(tx_status.digest_hex, tx_status.status) \
                             for tx_status in failed_this_round]
                 raise RuntimeError('Some transactions have failed: {}'.format(', '.join(failures)))
-
+            now = datetime.now()
             successful_this_round = [status for status in remaining_statuses if status.successful]
+            successful_this_round = [status for status in successful_this_round if (now-_get_or_set_default_time(hold_times, status.digest_hex, now))>=hold_state]
```

Comment:
This appears to add an artificial delay before allowing transactions to complete successfully - is it doing something more, or otherwise why is this useful?

Reply:
The TX can be "Executed" because it's part of the block generated, but when the block gets dumped with all the TX because of heavier chain. So executed TX later becomes unknown/.., by holding success state I can protect myself against this for some extent.